client.go

```diff
@@ -153,47 +147,52 @@ func (op *outgoingPacket) addEvent(ev *Event) (full bool) {
 	return (op.written + trailerSize) > op.maxUncompressed
 }
 
-func (c *UDPClient) Run(ctx context.Context) {
-	c.init()
+func (c *UDPClient) Submit(event *Event) {
+	c.mu.Lock()
+	defer c.mu.Unlock()
 
-	ticker := utils.NewJitteredTicker(c.FlushInterval)
-	var background errgroup.Group
-	defer func() { _ = background.Wait() }()
-	background.Go(func() error {
-		ticker.Run(ctx)
-		return nil
-	})
+	if c.closed {
+		return
+	}
 
-	p := c.newOutgoingPacket()
-
-	sendAndReset := func() {
-		_ = c.send(p, c.Addr)
-		p = c.newOutgoingPacket()
-	}
-
-	for {
-		select {
-		case em := <-c.submitQueue:
-			if p.addEvent(em) {
-				sendAndReset()
-			}
-		case <-ticker.C:
-			if p.events > 0 {
-				sendAndReset()
-			}
-		case <-ctx.Done():
-			left := len(c.submitQueue)
-			for i := 0; i < left; i++ {
-				if p.addEvent(<-c.submitQueue) {
-					sendAndReset()
-				}
-			}
-			if p.events > 0 {
-				_ = c.send(p, c.Addr)
-			}
-			return
-		}
+	if c.pending == nil {
+		c.pending = c.newOutgoingPacket()
+		c.flush = make(chan struct{}, 1)
+		c.active.Add(1)
+		go c.waitForFull(c.pending, c.flush)
```

Comment:
background goroutine?

Reply:
Yeah, sorry, for not being clear -- a long running background goroutine.